File: nomad-samples/python/basic-live-channel-1.1/account/refresh_token.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

from libraries import asyncio, json, requests

'''
 * Refresh Token
 *
 * @param {string} REFRESH_TOKEN | The refresh token
 *
 * @returns {string} Authentication token
'''
async def refresh_token(REFRESH_TOKEN):

    # Check for valid parameters
    if (not REFRESH_TOKEN):
        raise Exception("Refresh Token: The refresh token is invalid")


    # Create header for the request
    HEADERS = {
        'Content-Type': 'application/json'
    }

    # Build the payload BODY
    BODY = {
        "refreshToken": REFRESH_TOKEN
    }


    try:
        # Send the request
        RESPONSE = requests.post("PUBLIC_URL/account/refresh-token", headers= HEADERS, data= json.dumps(BODY))

        # Get the response
        INFO = json.loads(RESPONSE.text)

        if RESPONSE.status_code != 200:
            raise Exception()

        # Return the token
        return INFO["token"]

    except:
        api_exception_handler(RESPONSE, "Refresh Token failed")

File: nomad-samples/common/js/secure-video-1.2/account/refresh-token.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * Refresh Token
 *
 * @param {string} refreshToken | The refresh token
 *
 * @returns {string} Authorization token
 */
export default async function refreshToken() {
    // Get the refresh token
    const refreshToken = sessionStorage.getItem("refreshToken");

    // Check for valid parameters
    if (!refreshToken) {
        throw new Error("Refresh Token: The refresh token is invalid");
    }

    // Create header for the request
    const headers = new Headers();
    headers.append("Content-Type", "application/json");

    // Build the payload body
    const body = {
        refreshToken: refreshToken
    };

    // Send the request
    const response = await fetch(`${prjConstants.PUBLIC_URL}/account/refresh-token`, {
        method: "POST",
        headers: headers,
        body: JSON.stringify(body)
    });

    // Check for success
    if (response && response.ok) {
        // Get the response
        const refreshTokenResponse = await response.json();

        // Check for valid token
        if (refreshTokenResponse) {
            // Update the token
            sessionStorage.setItem("token", refreshTokenResponse.token);

            // Give feedback to the console
            console.info("Token refreshed: ", refreshTokenResponse);

            // Return the token
            return refreshTokenResponse.token;
        }
    }

    await apiExceptionHandler(response, "Refresh Token failed");
}
